### ISSUE TITLE
Unable to run tests with vmrunner on Mac

### ISSUE THREAD

**Issue by AndreasAakesson** [hioa-cs/IncludeOS#932]
I'm trying to run UDP intergration test with the supplied `test.py`, and I encounter two errors.

1. I've installed python with homebrew
```
$ which python
/usr/local/bin/python
```
test.py is using `#! /usr/bin/python`.
So even after I installed jsonschema `pip install jsonschema` it's not installed "correctly". Result:
```
$ sudo ./test.py 
Password:
includeos_src: /Users/andreas/dev/IncludeOS
Traceback (most recent call last):
  File "./test.py", line 11, in <module>
    import vmrunner
  File "/Users/andreas/dev/IncludeOS/test/vmrunner.py", line 10, in <module>
    import validate_test
  File "/Users/andreas/dev/IncludeOS/test/validate_test.py", line 2, in <module>
    from jsonschema import Draft4Validator, validators
ImportError: No module named jsonschema
```
Any good idea how to be flexible with which python is being used? Or should I just symlink apples one to homebrew?

2. Python is not using environment variables when calling `vmrunner.make()`, which makes it unable to compile on Mac since the necessary linker etc isn't declared (`LD_INC`).

Result:
```
$ sudo -E ./test.py 
includeos_src: /Users/andreas/dev/IncludeOS
[ WARNING ] WARNING: Environment varialble INCLUDEOS_HOME is not set. Trying default

==================== IncludeOS vmrunner initializing tests =====================
* <VMRunner>:  Validating test service

* <VMRunner>:  Loaded VM specification(s) from JSON
* <VMRunner>:  Found VM spec: 
{u'mem': 256, u'image': u'test_udp.img', u'net': [{u'device': u'virtio', u'mac': u'c0:01:0a:00:00:2a', u'backend': u'tap'}], u'cpu': u'host'}

* <VMRunner>:  Building test service with 'make' (params=[])
ld: unknown option: -nostdlib
make: *** [link_service] Error 1
Traceback (most recent call last):
  File "./test.py", line 45, in <module>
    vm.make().boot(20)
  File "/Users/andreas/dev/IncludeOS/test/vmrunner.py", line 253, in make
    res = subprocess.check_output(make)
  File "/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 574, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['make']' returned non-zero exit status 2
```

**Comment by AndreasAakesson** [comment]
Solution for issue 1:
```
-#! /usr/bin/python
+#! /usr/bin/env python
```

Needs to be updated in every script.

Issue 2 is no longer a issue with cmake. Tho every test need to define our toolchain file and have `project( ... )` added to it.

**Comment by AndreasAakesson** [comment]
Progress made in #958. Added the above into every CMakeLists.txt and test.py. Some scripts are using linux commands (`mkfs.fat`, `/proc/cpuinfo` to name a few) so they naturally won't work.

**Comment by AndreasAakesson** [comment]
This is fixed in #958. As stated above, some tests do not work on Mac, but that's a separate issue.
